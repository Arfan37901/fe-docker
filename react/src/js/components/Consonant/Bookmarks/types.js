import {
    bool,
    func,
    number,
} from 'prop-types';

// eslint-disable-next-line import/prefer-default-export
export const BookmarksType = {
    showBookmarks: bool,
    savedCardsCount: number,
    onClick: func.isRequired,
};
