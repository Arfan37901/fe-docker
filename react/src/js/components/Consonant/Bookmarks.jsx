import React from 'react';
import PropTypes from 'prop-types';

const Bookmarks = (props) => {
    const {
        selectedIco,
        unselectedIco,
        selected,
        onClick,
        qty,
    } = props;
    return (
        <button
            type="button"
            onClick={onClick}
            className={selected ? 'bookmarks bookmarks_selected' : 'bookmarks'
            }>
            <span className="bookmarks--ico-wrapper">
                {(selectedIco && unselectedIco) ?
                    <img
                        src={selected ? unselectedIco : selectedIco}
                        width="16"
                        alt=""
                        loading="lazy" /> :
                    <span className="bookmarks--ico" />
                }
                <span className="bookmarks--title">My favorites</span>
            </span>
            <span className="bookmarks--item-badge">{qty}</span>
        </button>
    );
};

export default Bookmarks;

Bookmarks.propTypes = {
    selectedIco: PropTypes.string,
    unselectedIco: PropTypes.string,
    selected: PropTypes.bool,
    onClick: PropTypes.func.isRequired,
    qty: PropTypes.number,
};

Bookmarks.defaultProps = {
    selectedIco: '',
    unselectedIco: '',
    selected: false,
    qty: 0,
};
