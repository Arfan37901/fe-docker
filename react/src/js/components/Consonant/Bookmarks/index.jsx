import React, { useMemo } from 'react';
import classNames from 'classnames';

import { selector } from './selectors';
import { BookmarksType } from './types';
import { useConfigSelector } from '../Helpers/hooks';

const defaultProps = {
    showBookmarks: false,
    savedCardsCount: 0,
};

const Bookmarks = ({ showBookmarks, onClick, savedCardsCount }) => {
    const {
        label,
        selectIcon,
        unselectIcon,
    } = useConfigSelector(selector);

    const icon = showBookmarks
        ? selectIcon
        : unselectIcon;

    const bookmarkClass = classNames({
        bookmarks: true,
        bookmarks_selected: showBookmarks,
    });

    const iconStyles = useMemo(() => ({ backgroundImage: `url(${icon})` }), [icon]);

    return (
        <button
            tabIndex="0"
            type="button"
            onClick={onClick}
            data-testid="bookmarks"
            className={bookmarkClass}>
            <span className="bookmarks__icon-wrapper">
                <span
                    style={iconStyles}
                    className="bookmarks--icon"
                    data-testid="bookmarks--icon" />
                <span className="bookmarks__label">{label}</span>
            </span>

            <span
                data-testid="bookmarks__badge"
                className="bookmarks--item-badge">
                {savedCardsCount}
            </span>
        </button>
    );
};

Bookmarks.propTypes = BookmarksType;
Bookmarks.defaultProps = defaultProps;

export default Bookmarks;
