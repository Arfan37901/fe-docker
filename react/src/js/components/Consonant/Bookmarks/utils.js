// eslint-disable-next-line import/prefer-default-export
export const selector = ({
    bookmarks: {
        leftFilterPanel: { selectBookmarksIcon, unselectBookmarksIcon },
        i18n: {
            leftFilterPanel: { filterTitle },
        },
    },
}) => ({
    label: filterTitle,
    selectIcon: selectBookmarksIcon,
    unselectIcon: unselectBookmarksIcon,
});
