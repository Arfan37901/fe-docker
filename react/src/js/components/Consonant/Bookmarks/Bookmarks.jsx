import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { useConfig } from '../../../utils/hooks';

const Bookmarks = (props) => {
    const {
        selected,
        onClick,
        qty,
    } = props;

    const getConfig = useConfig();
    const title = getConfig('bookmarks', 'i18n.leftFilterPanel.filterTitle');
    const selectedIco = getConfig('bookmarks', 'leftFilterPanel.selectBookmarksIcon');
    const unselectedIco = getConfig('bookmarks', 'leftFilterPanel.unselectBookmarksIcon');

    const src = selected ? selectedIco : unselectedIco;

    const iconStyles = useMemo(() => ({
        backgroundImage: src ? `url(${src})` : '',
    }));

    return (
        <button
            data-testid="bookmarks"
            type="button"
            onClick={onClick}
            className={selected ? 'bookmarks bookmarks_selected' : 'bookmarks'}
            tabIndex="0">
            <span className="bookmarks--ico-wrapper">
                <span
                    style={iconStyles}
                    className="bookmarks--ico"
                    data-testid="bookmarks--ico" />
                <span className="bookmarks--title">{title}</span>
            </span>
            <span data-testid="bookmarks--item-badge" className="bookmarks--item-badge">{qty}</span>
        </button>
    );
};

export default Bookmarks;

Bookmarks.propTypes = {
    selected: PropTypes.bool,
    onClick: PropTypes.func.isRequired,
    qty: PropTypes.number,
};

Bookmarks.defaultProps = {
    selected: false,
    qty: 0,
};
