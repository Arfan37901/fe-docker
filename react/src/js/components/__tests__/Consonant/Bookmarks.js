import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Bookmarks from '../../Consonant/Bookmarks';

import {
    COUNT_LIST,
    DEFAULT_PROPS,
} from '../constants/bookmarks';

import makeSetup from '../utils/settings';

Enzyme.configure({ adapter: new Adapter() });

const getSetup = makeSetup(Bookmarks, DEFAULT_PROPS);

describe('Consonant/Bookmarks', () => {
    test('should renders correctly different bookmarks count', () => {
        COUNT_LIST.forEach((count) => {
            const { wrapper } = getSetup({ qty: count });

            const badgeElement = wrapper.find('.bookmarks--item-badge');

            expect(badgeElement.text()).toEqual(String(count));
        });
    });

    test('should render unselectedIco', () => {
        const { props: { unselectedIco }, wrapper } = getSetup();

        const iconElement = wrapper.find('.bookmarks--ico');

        expect(iconElement).toHaveLength(1);
        expect(iconElement.get(0).props.style).toHaveProperty('backgroundImage', `url(${unselectedIco})`);
    });
    test('should render selectedIco', () => {
        const { props: { selectedIco }, wrapper } = getSetup({ selected: true });

        const iconElement = wrapper.find('.bookmarks--ico');

        expect(iconElement).toHaveLength(1);
        expect(iconElement.get(0).props.style).toHaveProperty('backgroundImage', `url(${selectedIco})`);
    });
    test('shouldn`t have style object when selected === true && selectedIco didnt exists', () => {
        const { wrapper } = getSetup({ selected: true, selectedIco: '' });

        const iconElement = wrapper.find('.bookmarks--ico');

        expect(iconElement).toHaveLength(1);
        expect(iconElement.get(0).props.style).toBeUndefined();
    });
    test('shouldn`t have style object when selected === false && unselectedIco didnt exists', () => {
        const { wrapper } = getSetup({ unselectedIco: '' });

        const iconElement = wrapper.find('.bookmarks--ico');

        expect(iconElement).toHaveLength(1);
        expect(iconElement.get(0).props.style).toBeUndefined();
    });

    describe('Check snapshots', () => {
        test('should renders correctly', () => {
            const { tree } = getSetup();

            expect(tree).toMatchSnapshot();
        });
    });

    describe('Interaction with UI', () => {
        test('should call onChange', () => {
            const { props: { onClick }, wrapper } = getSetup();

            wrapper.find('button').simulate('click');

            expect(onClick).toBeCalled();
            onClick.mockClear();
        });
    });
});
