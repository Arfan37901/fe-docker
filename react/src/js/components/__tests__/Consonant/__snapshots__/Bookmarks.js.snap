// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consonant/Bookmarks Check snapshots should renders correctly 1`] = `
<button
  className="bookmarks"
  onClick={[MockFunction]}
  type="button"
>
  <span
    className="bookmarks--ico-wrapper"
  >
    <span
      className="bookmarks--ico"
      style={
        Object {
          "backgroundImage": "url(path-to-unselectBookmarksIcon)",
        }
      }
    />
    <span
      className="bookmarks--title"
    >
      My favorites
    </span>
  </span>
  <span
    className="bookmarks--item-badge"
  >
    0
  </span>
</button>
`;
